test_util/azure_test_driver.py

```diff
@@ -55,20 +71,12 @@ def run():
     # tenant_id = os.environ.get('AZURE_TENANT_ID')
     # client_id = os.environ.get('AZURE_CLIENT_ID')
     # client_secret = os.environ.get('AZURE_CLIENT_SECRET')
-    group_name = 'tesing' + ''.join(random.choice('01234567890abcdef') for n in range(10))
+    group_name = 'testing' + ''.join(random.choice('01234567890abcdef') for n in range(10))
     deployment_name = 'deployment{}'.format(uuid.uuid4().hex)
 
     rmc = ResourceManagementClient(credentials, subscription_id)
 
     template_parameters = get_env_params()
-    if template_parameters.get('numberOfPrivateSlaves'):
-        assert template_parameters['numberOfPrivateSlaves']['value'] >= 2, 'Test requires at least 2 private slaves!'
```

Comment:
Hmm ok, still seems like an unnecessary tight coupling to me. The requirement for two agents can be explicitly captured by the CI job configuration, and integration test dependencies on the cluster spec are going to evolve over time, so I think it will be better if the test driver is generic and focuses on validating, deploying, and destroying ARM templates.